### PR TITLE
Update commands for matrix bot

### PR DESCRIPTION
**Pull request by prestist**

### CODE REVIEW

**Comment by prestist** [comment]
This is still a WIP and there are other parts I need to update. Not final.

But this should resolve issues found
https://github.com/coreos/fcos-meeting-action/issues/60
https://github.com/coreos/fcos-meeting-action/issues/48
https://github.com/coreos/fcos-meeting-action/issues/50

**Comment by dustymabe** [comment]
let me know when you are ready for review on this!

**Comment by prestist** [comment]
re: #60 
"the fcosmeetinghtml() should get updated to accept the new meeting html from the matrix bot. It kind of works right now but doesn't properly hyperlink things in the discussion forum post."

Looking at the existing structure of the logs, I think we might have lost some functionality moving to matrix. I may be mistaken but in the [past](https://meetbot-raw.fedoraproject.org/teams/fedora_coreos_meeting/fedora_coreos_meeting.2023-11-15-16.30.html) we had access to an actual link, [now](https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.html) we just get timestamps and user names. 

I am not sure, but I might be able to cobble something together.

**Comment by prestist** [comment]
@dustymabe I think its getting there, let me know if you see anything I might have missed

File: static/meeting-template.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Meeting Notes
 
-The meeting is held every week, meetings are at `16:30 UTC` on Wednesdays. The meeting is held in #fedora-meeting-1 on libera.chat. The meeting is logged and the logs are available [here](https://meetbot.fedoraproject.org/teams/fedora_coreos_meeting/)
+The meeting is held every week, meetings are at `16:30 UTC` on Wednesdays. The meeting is held in [#meeting-1:fedoraproject.org](https://matrix.to/#/#meeting-1:fedoraproject.org) on matrix. The meeting is logged and the logs are available [here](https://meetbot.fedoraproject.org/teams/fedora_coreos_meeting/)
```
Comment (dustymabe) [inline static/meeting-template.md:5]: I think here we can just put the full link and the matrix client will actually make it pretty for us (i.e. we don't need to do markdown for the link).

File: static/meeting-template.md [hunk 2]
```diff
@@ -18,8 +18,8 @@ Please see [meeting-people.txt](https://github.com/coreos/fedora-coreos-tracker/
 
 <details>
 <summary>Meeting host Requirements</summary>
-The meeting host needs to have IRC configured and setup on their machine.
-The host needs to have access to the following channels on libera.chat
+The meeting host needs to have matrix configured with an account on the [fedora matrix server](https://chat.fedoraproject.org/#/room/#coreos:fedoraproject.org).
+The host needs to have access to the following channels on matrix
     - `#fedora-meeting-1`
     - `#fedora-coreos`
```
Comment (dustymabe) [inline static/meeting-template.md:24]: We should probably make these links to the channels in matrix and make them markdown:

- [#coreos:fedoraproject.org](https://matrix.to/#/#coreos:fedoraproject.org)
- [#meeting-1:fedoraproject.org](https://matrix.to/#/#meeting-1:fedoraproject.org)

File: static/meeting-template.md [hunk 3]
```diff
@@ -18,8 +18,8 @@ Please see [meeting-people.txt](https://github.com/coreos/fedora-coreos-tracker/
 
 <details>
 <summary>Meeting host Requirements</summary>
-The meeting host needs to have IRC configured and setup on their machine.
-The host needs to have access to the following channels on libera.chat
+The meeting host needs to have matrix configured with an account on the [fedora matrix server](https://chat.fedoraproject.org/#/room/#coreos:fedoraproject.org).
```
Comment (dustymabe) [inline static/meeting-template.md:21]: The meeting host needs to have a matrix account (an example is an account on the [fedora matrix server](https://chat.fedoraproject.org/#/room/#coreos:fedoraproject.org)).

File: static/meeting-template.md [hunk 4]
```diff
@@ -52,12 +52,12 @@ At least 5 people must vote, or 51% of the WG membership, whichever is less. Vot
 
 ## Meeting Steps
 
-1. Join channel #fedora-meeting-1 on libera.chat copy these commands and paste them in the channel
+1. Join channel [#fedora-meeting-1](https://matrix.to/#/#coreos:fedoraproject.org) on matrix, copy these commands and paste them in the channel
```
Comment (dustymabe) [inline static/meeting-template.md:55]: Maybe let's use the matrix formatted name here:

```
[#coreos:fedoraproject.org](https://matrix.to/#/#coreos:fedoraproject.org)
```

same for line 60 below

Comment (dustymabe) [reply static/meeting-template.md:55]: and line 70

File: static/meeting-template.md [hunk 5]
```diff
@@ -120,22 +120,39 @@ At least 5 people must vote, or 51% of the WG membership, whichever is less. Vot
...
+                curl -s "$SUMMARY_URL" | sed -E -e "$sed_commands" | sed '/<head>/,/<\/head>/d' | sed 's/TOPIC://g'
+            }
+
+            # Example usage
+            BASE_URL="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/"
+            SUMMARY_URL="${BASE_URL}fedora-coreos-meeting.2024-01-10-16.35.html"
+            LOG_URL="${BASE_URL}fedora-coreos-meeting.2024-01-10-16.35.log.html"
+
+            # Generate HTML with hyperlinks, delete <head> contents, and echo the result
+            fcosmeetinghtml "$SUMMARY_URL" "$LOG_URL"
```
Comment (dustymabe) [inline static/meeting-template.md:154]: We should break this out so one can just click the copy button in the top right corner of this box to copy/paste the function into their shell.

File: static/meeting-template.md [hunk 6]
```diff
@@ -122,22 +120,39 @@ At least 5 people must vote, or 51% of the WG membership, whichever is less. Vot
...
-                url=$(echo "$url" | sed "s|meetbot.fedoraproject.org|meetbot-raw.fedoraproject.org|")
-            fi
-            # Take the html and delete the <head> contents (doesn't render correctly)
-            # and also substitute in the base url to make relative URLs absolute.
-            curl --silent $url                                | \
-                sed -z 's|<head>.*</head>||'                  | \
-                sed "s|href='fedora|href='${baseurl}/fedora|" | \
-                sed "s|href=\"fedora|href=\"${baseurl}/fedora|"
-        }
+            fcosmeetinghtml() {
```
[the remark this answers is not in the document]
Comment (prestist) [reply static/meeting-template.md:124]: Ah, I did not get around to pasting it in a forum post, I will take a look.

File: static/meeting-template.md [hunk 7]
```diff
@@ -120,22 +120,39 @@ At least 5 people must vote, or 51% of the WG membership, whichever is less. Vot
...
+                SUMMARY_URL="$1"
+                LOG_URL="$2"
```
[the remark this answers is not in the document]
Comment (prestist) [reply static/meeting-template.md:126]: 100% agree, was getting burned out on working on the function, will circle back

File: static/meeting-template.md (hunk 5 above)
Comment (prestist) [reply static/meeting-template.md:154]: ah good point.

File: static/meeting-template.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (prestist) [reply static/meeting-template.md:124]: Hmm, I did not see that,

When copying the whole block The output is:

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html>
<body>
    <h1>#meeting-1:fedoraproject.org: fedora_coreos_meeting</h1>
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-1">    <span class="details">Meeting started by @spresti:fedora.im at 16:35:03 UTC</span></a>
    <br><br>
    <h3>Meeting summary</h3>
    <ol>
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-4">                    <li><b></b><b class="TOPIC">roll call</b> <span class="details">(@spresti:fedora.im, 16:35:16)</span></a></li>
                    <ol type="a">
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-24"></ol>                    <li><b></b><b class="TOPIC">Action items from last meeting</b> <span class="details">(@spresti:fedora.im, 16:38:59)</span></a></li>
                    <ol type="a">
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-27"></ol>                    <li><b></b><b class="TOPIC">there are no action items from the last meeting.</b> <span class="details">(@spresti:fedora.im, 16:40:16)</span></a></li>
                    <ol type="a">
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-30"></ol>                    <li><b></b><b class="TOPIC">New Package Request: mtr</b> <span class="details">(@spresti:fedora.im, 16:41:06)</span></a></li>
                    <ol type="a">
                <li>LINK: https://github.com/coreos/fedora-coreos-tracker/issues/1644 (@spresti:fedora.im, 16:41:17)</li>
                <li>AGREED: the basic functionality of mtr already exists in FCOS through tracepath. Additionally mtr can be installed and ran in a toolbox container. So we currently do not see the need to add this package to the base for everyone. (@spresti:fedora.im, 16:53:51)</li>
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-58"></ol>                    <li><b></b><b class="TOPIC">aarch64 failing to upgrade with /boot filesystem full</b> <span class="details">(@spresti:fedora.im, 16:54:22)</span></a></li>
                    <ol type="a">
                <li>LINK: https://github.com/coreos/fedora-coreos-tracker/issues/1637 (@spresti:fedora.im, 16:54:34)</li>
                <li>AGREED: continue with the proposed mitigation/fix for this is in https://github.com/coreos/fedora-coreos-tracker/issues/1637#issuecomment-1878186381 which we are actively executing. (@spresti:fedora.im, 17:00:27)</li>
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-67"></ol>                    <li><b></b><b class="TOPIC">tracker: Fedora 40 changes considerations</b> <span class="details">(@spresti:fedora.im, 17:00:55)</span></a></li>
                    <ol type="a">
                <li>LINK: https://github.com/coreos/fedora-coreos-tracker/issues/1626 (@spresti:fedora.im, 17:01:11)</li>
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-79"></ol>                    <li><b></b><b class="TOPIC">Open Floor</b> <span class="details">(@spresti:fedora.im, 17:06:53)</span></a></li>
                    <ol type="a">
            </ol>
    </ol>
    <br/>
    <br/>
<a href="https://meetbot-raw.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-10/fedora-coreos-meeting.2024-01-10-16.35.log.html#l-99">    <span class="details">Meeting ended at 17:19:10 UTC</span></a>
    <br/>
    <br/>



<h3>Action items</h3>
<ol>
    <li>(none)</li>
</ol>
<br><br>



<h3>People present (lines said)</h3>
<ol>
<li>@spresti:fedora.im (42)</li>
<li>@siosm:matrix.org (21)</li>
<li>@dustymabe:matrix.org (18)</li>
<li>@zodbot:fedora.im (7)</li>
<li>@fifofonix:matrix.org (4)</li>
<li>@jlebon:fedora.im (4)</li>
<li>@meetbot:fedora.im (2)</li>
<li>@marmijo:fedora.im (1)</li>
</ol>
<br><br>



</body></html>%

[the remark this answers is not in the document]
Comment (prestist) [reply static/meeting-template.md:124]: Not sure whats up with the '%' but otherwise looks good.

[the remark this answers is not in the document]
Comment (prestist) [reply static/meeting-template.md:124]: Interesting markdown in github is removing the links, They are there...

File: static/meeting-template.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (prestist) [reply static/meeting-template.md]: Ok its not pretty but gave it a swing.

File: static/meeting-template.md [hunk 8]
```diff
@@ -67,13 +67,11 @@ At least 5 people must vote, or 51% of the WG membership, whichever is less. Vot
         If you don't want to be pinged remove your name from this file: https://github.com/coreos/fedora-coreos-tracker/blob/main/issue_template/meeting-template.md
```
Comment (dustymabe) [inline static/meeting-template.md:67]: the line above needs to say:

```
FCOS community meeting in https://matrix.to/#/#meeting-1:fedoraproject.org
```

**Comment by dustymabe** [comment]
I'm still getting a lot of extra output.. here is what I am doing:

```
fcosmeetinghtml https://meetbot.fedoraproject.org/meeting-1_matrix_fedoraproject-org/2024-01-17/fedora-coreos-meeting.2024-01-17-16.29.html
```

Then copying that into a forum post.

**Comment by prestist** [comment]
> Then copying that into a forum post.

hmm, I wonder if there is something system based which is causing that. Do you have something in your .rc file? 

Try naming the function something else and seeing if there is less output?

**Review by dustymabe — APPROVED** [review]
LGTM - what we have is an improvement and we'll continue to tweak it as we go

**Pull request merged**